### ISSUE TITLE
Unexpected events.tsv file missing warning

### ISSUE THREAD

**Issue by VisLab** [bids-standard/bids-validator#213]
I am updating some datasets that are on openNeuro but were validated with much earlier versions of the validator.  I am getting the following unexpected (and I think incorrectly given) warning:

```code
  [WARNING] EVENTS_TSV_MISSING Task scans should have a corresponding 'events.tsv' file.
If this is a resting state scan you can ignore this warning or rename the task to include the word "rest".

                /sub-001/eeg/sub-001_task-AuditoryVisualShift_run-01_channels.tsv
                /sub-001/eeg/sub-001_task-AuditoryVisualShift_run-01_eeg.fdt

                154 more files with the same issue
```

The validator is: https://jsr.io/@bids/validator/2.0.7/src/bids-validator.ts.  This happens on both the online validator and running locally directly using deno.  The validator recognizes the `events.tsv` files since it also gives a valid warning:

```code
[WARNING] TSV_ADDITIONAL_COLUMNS_UNDEFINED A TSV file has extra columns which are not defined in its associated JSON sidecar
                sample
                /sub-001/eeg/sub-001_task-AuditoryVisualShift_run-01_events.tsv
                /sub-002/eeg/sub-002_task-AuditoryVisualShift_run-01_events.tsv

                50 more files with the same issue
```

**Comment by suxpert** [comment]
I've got the same issue on a custom dataset with the online validator. The BIDS contains files like this:
```
bidsroot/sub-338
├── anat
│   ├── sub-338_acq-highres_T1w.json
│   └── sub-338_acq-highres_T1w.nii.gz
├── fmap
│   ├── sub-338_magnitude1.json
│   ├── sub-338_magnitude1.nii.gz
│   ├── sub-338_magnitude2.json
│   ├── sub-338_magnitude2.nii.gz
│   ├── sub-338_phasediff.json
│   └── sub-338_phasediff.nii.gz
└── func
    ├── sub-338_task-norm_run-01_bold.json
    ├── sub-338_task-norm_run-01_bold.nii.gz
    ├── sub-338_task-norm_run-01_events.tsv
    ├── sub-338_task-norm_run-02_bold.json
    ├── sub-338_task-norm_run-02_bold.nii.gz
    ├── sub-338_task-norm_run-02_events.tsv
    ├── sub-338_task-norm_run-03_bold.json
    ├── sub-338_task-norm_run-03_bold.nii.gz
    ├── sub-338_task-norm_run-03_events.tsv
    ├── sub-338_task-prob_run-01_bold.json
    ├── sub-338_task-prob_run-01_bold.nii.gz
    ├── sub-338_task-prob_run-01_events.tsv
    ├── sub-338_task-prob_run-02_bold.json
    ├── sub-338_task-prob_run-02_bold.nii.gz
    ├── sub-338_task-prob_run-02_events.tsv
    ├── sub-338_task-prob_run-03_bold.json
    ├── sub-338_task-prob_run-03_bold.nii.gz
    └── sub-338_task-prob_run-03_events.tsv
```

And the online [validator](https://bids-standard.github.io/bids-validator/) (showed `Schema version: 1.0.4`) gives tons of warnings (about `SIDECAR_KEY_RECOMMENDED`), and with the following **unexpected** warning:
```
Task scans should have a corresponding 'events.tsv' file. If this is a resting state scan you can ignore this warning or rename the task to include the word "rest".

    /sub-338/func/sub-338_task-norm_run-01_bold.nii.gz
    /sub-338/func/sub-338_task-norm_run-02_bold.nii.gz
    /sub-338/func/sub-338_task-norm_run-03_bold.nii.gz
    /sub-338/func/sub-338_task-prob_run-01_bold.nii.gz
    /sub-338/func/sub-338_task-prob_run-02_bold.nii.gz
    /sub-338/func/sub-338_task-prob_run-03_bold.nii.gz
   ...
```

The validator DO recognize those `events.tsv` files, since if I modify the `/events.json`, it would complain about `TSV_ADDITIONAL_COLUMNS_UNDEFINED`, the same as in @VisLab 's case.

**Comment by monique2208** [comment]
We have also encountered this issue with the latest version. It is a bit of a problem because we upgrade the events_tsv_missing to an error in our repository

**Comment by ericearl** [comment]
I had this happen too and hadn't gotten back here before to say so. I believe it was either MEG events TSVs or fMRI events TSVs, but I can verify if needed.

**Comment by effigies** [comment]
This will be fixed in https://github.com/bids-standard/bids-specification/pull/2166
